src/Setup/WixBA/RootView.xaml.cs

```diff
@@ -35,7 +36,37 @@ public RootView(RootViewModel viewModel)
         /// <param name="e"></param>
         private void Background_MouseLeftButtonDown(object sender, MouseButtonEventArgs e)
         {
-            this.DragMove();
+            if (MouseButtonState.Pressed == Mouse.LeftButton)
+            {
+                this.DragMove();
+                e.Handled = true;
+            }
+        }
+ 
+        /// <summary>
+        /// If user clicks on a no-input control (i.e. a TextBlock, etc.) route event to here to avoid  InvalidOperationException
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void RootView_IgnoreClick(object sender, MouseButtonEventArgs e)
+        {
+            //e.Handled = true;
+        }
+
+        /// <summary>
+        /// Event is fired when the window is closing.
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void Window_Closing(object sender, System.ComponentModel.CancelEventArgs e)
+        {
+            RootViewModel rvm = this.DataContext as RootViewModel;
+            if ((null != rvm) && (InstallationState.Applying == rvm.InstallState))
+            {
+                rvm.Canceled = true;
```

Comment:
It does prompt on a Cancel.

On Mon, Aug 1, 2016 at 12:03 AM, Sean Hall <notifications@github.com> wrote:

> In src/Setup/WixBA/RootView.xaml.cs
> <https://github.com/wixtoolset/wix4/pull/208#discussion_r72924455>:
>
> > +        private void RootView_IgnoreClick(object sender, MouseButtonEventArgs e)
> > +        {
> > +            //e.Handled = true;
> > +        }
> > +
> > +        /// <summary>
> > +        /// Event is fired when the window is closing.
> > +        /// </summary>
> > +        /// <param name="sender"></param>
> > +        /// <param name="e"></param>
> > +        private void Window_Closing(object sender, System.ComponentModel.CancelEventArgs e)
> > +        {
> > +            RootViewModel rvm = this.DataContext as RootViewModel;
> > +            if ((null != rvm) && (InstallationState.Applying == rvm.InstallState))
> > +            {
> > +                rvm.Canceled = true;
>
> I can't remember if WixBA prompts when the user clicks the Cancel button.
> If so, we also should prompt here. If not, there's probably a bug somewhere
> asking us to prompt.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/wixtoolset/wix4/pull/208/files/28721ab8e4985a5641b3efd7975a67cecd8f8d4a#r72924455>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHyCLcB1zgGVc9a6I2LoopfxYA8oKv4vks5qbX4QgaJpZM4JVm8P>
> .
>